src/MSONUtility.h

```diff
@@ -121,26 +121,28 @@ namespace mson {
      *
      * \param subject String which represents the type name
      * \param typeName MSON Type Name
+     * \param notBaseType If true, will be parsed as a symbol
      */
     inline void parseTypeName(const std::string& subject,
-                              TypeName& typeName) {
+                              TypeName& typeName,
+                              bool notBaseType = false) {
```

Comment:
the name is not really fortunate – having `not` in the name and then `!` condition it is really hard to read...